### PR TITLE
CSP-1344 enter employer details paye and aorn

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+[Route("/{ukprn}/addEmployer/searchByPaye", Name = RouteNames.AddEmployerSearchByPaye)]
+public class SearchByPayeController(IOuterApiClient _outerApiClient, ISessionService _sessionService, IValidator<SearchByPayeSubmitViewModel> _validator) : Controller
+{
+    public const string ViewPath = "~/Views/AddEmployer/SearchByPaye.cshtml";
+
+
+    [HttpGet]
+    public IActionResult Index([FromRoute] int ukprn)
+    {
+        var viewModel = GetViewModel(ukprn);
```
Comment (DevenKShah) [inline src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs:25]: Move this down so we initiate view model only if session check passes.

File: src/SFA.DAS.Provider.PR.Web.UnitTests/Validators/SearchByPayeSubmitViewModelValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    [TestCase("111/12", "1234567890123")]
+    [TestCase("111/ABC", "1234567890123")]
+    [TestCase("111/ABCD", "1234567890123")]
+    [TestCase("111/ABCDE", "1234567890123")]
+    [TestCase("111/ABCDEF", "1234567890123")]
+    [TestCase("111/ABCDEFG", "1234567890123")]
+    [TestCase("111/ABCDEFGH", "1234567890123")]
+    [TestCase("111/ABCDEFGHI", "1234567890123")]
+    [TestCase("111/ABCDEFGHIJ", "1234567890123")]
+    public void ValidDetailsInModel_IsValid(string payeRef, string aorn)
```
Comment (DevenKShah) [inline src/SFA.DAS.Provider.PR.Web.UnitTests/Validators/SearchByPayeSubmitViewModelValidatorTests.cs:18]: I got confused on the intension of the test. This seems like testing valid scenario for PAYE as AORN is constant here. I would just write this test specifically for valid PAYE cases. Also I do not see point of incrementing test for valid length, border line tests would suffice.

**Review by DevenKShah — APPROVED** [review]

File: src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs:25]: ok

File: src/SFA.DAS.Provider.PR.Web.UnitTests/Validators/SearchByPayeSubmitViewModelValidatorTests.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.Provider.PR.Web.UnitTests/Validators/SearchByPayeSubmitViewModelValidatorTests.cs:18]: ok

**Review by DevenKShah — DISMISSED** [review]

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**